### PR TITLE
Adding Codeowners

### PR DESCRIPTION
**Pull request by mustermeiszer**
# Description
This PR adds a CODEOWNER file to our repository. This alone does not enable anything but when the repository enforces code owners to do a review, only with a review of them PRs can be merged. 

- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+pallets/liquidity-rewards/* @lemunozm
+
+### Nft-sales related logic
+pallets/nft-sales/* @NunoAlexandre
+
+### Pallet Keystore
+pallets/keystore/* @cdamian
+
+# Changes to libs
+libs/* @branan @mikiquantum @mustermeiszer
```
Comment (lemunozm) [inline .github/CODEOWNERS:30]: Here could have sections where any other than these three can review it (in my case, fees, ensure, rewards modules, i.e). Is it needed to express that level too?

Comment (mustermeiszer) [reply .github/CODEOWNERS:30]: Yes, that would be needed. Tell me what you want and I add it

Comment (lemunozm) [reply .github/CODEOWNERS:30]: `fees`, `ensure`, `rewards` modules for me in `cfg_traits` 👍🏻

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+#### ensure module
+libs/traits/src/ops.rs @lemunozm
+
+# Changes to runtime
+runtime/* @branan @mikiquantum @mustermeiszer
+
+## Changes to folders inside runtime
+### Changes to runtime/integration-tests
+runtime/integration-tests/* @mustermeiszer
```
Comment (NunoAlexandre) [inline .github/CODEOWNERS:49]: I would prefer to be included to this rule as well. Currently the integration tests are divided by pools and xcm tests, so imv it makes sense to have us both on the root level. wdyt?

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -0,0 +1,48 @@
+# Centrifudge Code Owners
+#
+# Order is important; the last matching pattern takes the most
+
+# These owners will be the default owners for everything in
+# the repo. Unless a later match takes precedence,
+*       @branan @mikiquantum @mustermeiszer @offerijns
```
Comment (NunoAlexandre) [inline .github/CODEOWNERS:7]:
```suggestion
* @branan @mikiquantum @mustermeiszer @offerijns
```

File: .github/CODEOWNERS [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+# Changes to the service of our chain.
+src/*    @mikiquantum
```
Comment (NunoAlexandre) [inline .github/CODEOWNERS:10]:
```suggestion
src/* @mikiquantum
```

File: .github/CODEOWNERS [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+# Changes to pallets
+pallets/*  @branan @mikiquantum @mustermeiszer
```
Comment (NunoAlexandre) [inline .github/CODEOWNERS:13]:
```suggestion
pallets/* @branan @mikiquantum @mustermeiszer
```

File: .github/CODEOWNERS [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+# Changes to runtime
+runtime/* @branan @mikiquantum @mustermeiszer
+
+## Changes to folders inside runtime
+### Changes to runtime/integration-tests
+runtime/integration-tests/* @mustermeiszer
+
+### Changes to runtime/integration-tests/xcm/
+runtime/integration-tests/xcm/* @NunoAlexandre
```
Comment (NunoAlexandre) [inline .github/CODEOWNERS:52]:
```suggestion
runtime/integration-tests/xcm/* @NunoAlexandre

```

File: .github/CODEOWNERS (hunk 3 above)
Comment (NunoAlexandre) [inline .github/CODEOWNERS:7]: Btw, @offerijns usually doesn't have the bandwidth to respond to PRs in a short timeline. The other engineer in the Europe timezone is yourself, also not at full capacity. I would suggest adding myself here for that reason.

Comment (offerijns) [reply .github/CODEOWNERS:7]: Just sharing my two cents on this: I think it's important that we work on prioritising reviews above other work, and this is the right fix for that problem (and from what I've seen, this is going increasingly better in the protocol circle). But, as an example, if there is a PR that is touching the `pool-system` pallet, I think at least 1 of Branan/Frederik/me should review that (and the same logic applies to other pallets for other people, although `pool-system` is probably most critical). I don't think this should be reviewed solely by other people, it should imo always include someone from that list, even if this means waiting another day until they are available.

I'm wondering if this means we should not even remove this asterisk rule, and just have specific rules covering every individual folder. I checked the code owners docs, if a file is not covered by any of the rules (e.g. a change to just the README), then no code owner review would be required and any contributor with write access (i.e. the entire protocol circle) can review.

Comment (mustermeiszer) [reply .github/CODEOWNERS:7]: My main reasoning is, that the asterix prevents us from missing a rule. I.e. We need to adapt the file everytime we add a new pallet and define a owner. 

But this is just another argument, I am not against removing it. Just to take into account.

File: .github/CODEOWNERS (hunk 6 above)
Comment (mustermeiszer) [reply .github/CODEOWNERS:51]: I will still keep you in that line, additionally, as this will notify you for review. Wdyt?

Comment (NunoAlexandre) [reply .github/CODEOWNERS:51]: Sounds good, thanks 🙏

File: .github/CODEOWNERS [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+# Changes to .github
+.github/* @mikiquantum @mustermeiszer @branan @NunoAlexandre @offerjins
+
+# Changes to ci
+ci/* @mikiquantum @mustermeiszer @branan @NunoAlexandre @offerjins
+
+# Changes to the service of our chain.
+src/* @mikiquantum @mustermeiszer @branan @NunoAlexandre
+
+# Changes to chain-specs
```
Comment (mustermeiszer) [inline .github/CODEOWNERS:12]: Do we actually need that line?

File: .github/CODEOWNERS [hunk 8]
```diff
@@ -0,0 +1,71 @@
+# Centrifudge Code Owners
+
+# Changes to .github
+.github/* @mikiquantum @mustermeiszer @branan @NunoAlexandre @offerijns
+
+# Changes to ci
```
Comment (mustermeiszer) [inline .github/CODEOWNERS:6]: Do we actually need that?

File: .github/CODEOWNERS [hunk 9]
```diff
@@ -0,0 +1,71 @@
+# Centrifudge Code Owners
+
+# Changes to .github
+.github/* @mikiquantum @mustermeiszer @branan @NunoAlexandre @offerijns
```
Comment (mustermeiszer) [inline .github/CODEOWNERS:4]: Do we actually need that?

**Review by mustermeiszer — COMMENTED** [review]
Some thoughts that need comments from the others.

File: .github/CODEOWNERS (hunk 8 above)
Comment (mustermeiszer) [reply .github/CODEOWNERS:6]: This is referring to ci folder changes

File: .github/CODEOWNERS (hunk 9 above)
Comment (offerijns) [inline .github/CODEOWNERS:4]: I don't know much about how the workflows are setup.
```suggestion
.github/* @mikiquantum @mustermeiszer @branan @NunoAlexandre
```

File: .github/CODEOWNERS [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+
+# Changes to ci
+ci/* @mikiquantum @mustermeiszer @branan @NunoAlexandre @offerijns
```
Comment (offerijns) [inline .github/CODEOWNERS:7]: Similarly
```suggestion
ci/* @mikiquantum @mustermeiszer @branan @NunoAlexandre
```

File: .github/CODEOWNERS [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+
+# Changes to the service of our chain.
+src/* @mikiquantum @mustermeiszer @branan @NunoAlexandre
+
+# Changes to chain-specs
+res/* @mikiquantum @mustermeiszer @branan @NunoAlexandre
+
+# Changes to specific pallets
+## Connectors related logic
+pallets/connectors/* @NunoAlexandre
```
Comment (offerijns) [inline .github/CODEOWNERS:17]: I think I can be a backup for this.
```suggestion
pallets/connectors/* @NunoAlexandre @offerijns
```

File: .github/CODEOWNERS [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+
+## Rewards related logic
+pallets/rewards/* @lemunozm
+pallets/liquidity-rewards/* @lemunozm
```
Comment (offerijns) [inline .github/CODEOWNERS:21]:
```suggestion
pallets/rewards/* @lemunozmn @mikiquantum
pallets/liquidity-rewards/* @lemunozm @mikiquantum
```

File: .github/CODEOWNERS [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+
+## Nft-sales related logic
+pallets/nft-sales/* @NunoAlexandre
+
+## Pallet Keystore
+pallets/keystore/* @cdamian
```
Comment (offerijns) [inline .github/CODEOWNERS:27]:
```suggestion
pallets/keystore/* @cdamian  @mikiquantum
```

File: .github/CODEOWNERS (hunk 3 above)
Comment (mustermeiszer) [reply .github/CODEOWNERS]: I think I see your points now @NunoAlexandre and @offerijns. I adapted the owner file.

**Review by NunoAlexandre — APPROVED** [review]
LGTM! Thanks @mustermeiszer

File: .github/CODEOWNERS [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+pallets/permissions/* @mustermeiszer
+
+## Pallet Restricted-Tokens
+pallets/restricted-tokens/* @mustermeiszer
+
+# Changes to libs
+## Changes to specific libraries
+### cfg-traits crate
+#### rewards module
+libs/traits/src/rewards.rs @lemunozm
```
Comment (lemunozm) [inline .github/CODEOWNERS:54]: Not sure If we need an entry for the rest of the `libs/*`?

Comment (mustermeiszer) [reply .github/CODEOWNERS:54]: I thought this would be a place that everybody can already own. WDYT?

Comment (lemunozm) [reply .github/CODEOWNERS:54]: I agree 👍🏻

**Review by lemunozm — APPROVED** [review]

**Pull request merged**